WeCantSpell.Hunspell/WordList.cs

```diff
@@ -161,6 +156,197 @@ private void ApplyRootOutputConversions(ref SpellCheckResult result)
+    /// <remarks>
+    /// Changes made to this dictionary instance will not be saved.
+    /// </remarks>
+    public bool Add(string word)
+    {
+        return Add(word, FlagSet.Empty, MorphSet.Empty, WordEntryOptions.None);
+    }
+
+    /// <summary>
+    /// Adds a root word to this in-memory dictionary.
+    /// </summary>
+    /// <param name="word">The root word to add.</param>
+    /// <param name="flags">The flags associated with the root <paramref name="word"/> detail entry.</param>
+    /// <param name="morphs">The morphs associated with the root <paramref name="word"/> detail entry.</param>
+    /// <param name="options">The options associated with the root <paramref name="word"/> detail entry.</param>
+    /// <returns><c>true</c> when a root is added, <c>false</c> otherwise.</returns>
+    /// <remarks>
+    /// Changes made to this dictionary instance will not be saved.
+    /// </remarks>
+    public bool Add(string word, FlagSet flags, IEnumerable<string> morphs, WordEntryOptions options)
+    {
+        return Add(word, flags, MorphSet.Create(morphs), options);
+    }
+
+    /// <summary>
+    /// Adds a root word to this in-memory dictionary.
+    /// </summary>
+    /// <param name="word">The root word to add.</param>
+    /// <param name="flags">The flags associated with the root <paramref name="word"/> detail entry.</param>
+    /// <param name="morphs">The morphs associated with the root <paramref name="word"/> detail entry.</param>
+    /// <param name="options">The options associated with the root <paramref name="word"/> detail entry.</param>
+    /// <returns><c>true</c> when a root is added, <c>false</c> otherwise.</returns>
+    /// <remarks>
+    /// Changes made to this dictionary instance will not be saved.
+    /// </remarks>
+    public bool Add(string word, FlagSet flags, MorphSet morphs, WordEntryOptions options)
+    {
+        return Add(word, new WordEntryDetail(flags, morphs, options));
+    }
+
+    /// <summary>
+    /// Adds a root word to this in-memory dictionary.
+    /// </summary>
+    /// <param name="word">The root word to add details for.</param>
+    /// <param name="detail">The details to associate with the root <paramref name="word"/>.</param>
+    /// <returns><c>true</c> when a root is added, <c>false</c> otherwise.</returns>
+    /// <remarks>
+    /// Changes made to this dictionary instance will not be saved.
+    /// </remarks>
+    public bool Add(string word, WordEntryDetail detail)
+    {
+        return Add(_entriesByRoot, Affix, word, detail);
+    }
+
+    private static bool Add(TextDictionary<WordEntryDetail[]> entries, AffixConfig affix, string word, WordEntryDetail detail)
+    {
+        if (affix.IgnoredChars.HasItems)
+        {
+            word = affix.IgnoredChars.RemoveChars(word);
+        }
+
+        if (affix.ComplexPrefixes)
+        {
+            word = word.GetReversed();
+        }
+
+        ref var details = ref entries.GetOrAddValueRef(word)!;
```

Comment:
Mention the behavior change of Add on the Builder for `IgnoredChars` and `ComplexPrefixes`.